### PR TITLE
Add config, build, and publish files for SQL Server 2022 Preview

### PR DESCRIPTION
**Pull request by jjerger**
Adds daisy workflows to build SQL Server 2022 Preview on Windows Server 2022

### CODE REVIEW

File: daisy_workflows/build-publish/sqlserver/sql-2022-preview-windows-2022-dc-uefi.publish.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  {{$time := trimPrefix .publish_version "v"}}
+  "Images": [
+    {
+      "Prefix": "sql-2022-preview-windows-2022-dc",
+      "Family": "sql-ent-2022-win-2022",
+      "Description": "Microsoft, SQL Server 2022 Preview, on Windows Server 2022 Datacenter, x64 built on {{$time}}, supports Shielded VM features",
+      "Architecture": "X86_64",
+      "Licenses": [
+        {{if eq .environment "staging" -}}
+        "projects/windows-sql-cloud/global/licenses/sql-server-2022-express",
```
Comment (jjerger) [inline daisy_workflows/build-publish/sqlserver/sql-2022-preview-windows-2022-dc-uefi.publish.json:35]: Is this inappropriate? I didn't see any sql licenses at all in bct-staging-functional.

Comment (bkatyl) [reply daisy_workflows/build-publish/sqlserver/sql-2022-preview-windows-2022-dc-uefi.publish.json:35]: No, you'll need to create a sql-server-2022-express in bct-staging-functional

Comment (jjerger) [reply daisy_workflows/build-publish/sqlserver/sql-2022-preview-windows-2022-dc-uefi.publish.json:35]: Turns out I'd already done this, so I've updated it.

Comment (bkatyl) [reply daisy_workflows/build-publish/sqlserver/sql-2022-preview-windows-2022-dc-uefi.publish.json:35]: I don't see any update to the PR, did you forget to push a change?

Comment (jjerger) [reply daisy_workflows/build-publish/sqlserver/sql-2022-preview-windows-2022-dc-uefi.publish.json]: That I did. Pushed it now.

**Review by bkatyl — APPROVED** [review]

**Pull request merged**